src/main/scala/viper/silver/parser/Translator.scala

```diff
@@ -265,6 +265,32 @@ case class Translator(program: PProgram) {
     }
   }
 
+  def fieldAssignStmt(errorNode: PNode, targets: Seq[PAssignTarget], assign: Seq[LocalVar] => Stmt): Stmt = {
+    val tTargets = targets map exp
+    val ts = tTargets.zipWithIndex.map {
+      case (lv: LocalVar, _) => (None, lv)
+      case (fa: FieldAccess, i) => {
+        val rcv = LocalVar(s"__silver_rcv_$i", fa.typ)(fa.rcv.pos)
+        val tgt = LocalVar(s"__silver_tgt_$i", fa.typ)(fa.pos)
+        val rcvFa = FieldAccess(rcv, fa.field)(fa.pos, fa.info, fa.errT)
+        (Some((rcv, fa.rcv, rcvFa, tgt)), tgt)
+      }
+      case _ => sys.error(s"Found invalid target of assignment")
+    }
+    val assn = assign(ts.map(_._2))
+    val tmps = ts.flatMap(_._1)
+    if (tmps.isEmpty)
+      return assn
+    if (!Consistency.noDuplicates(tmps.map(_._3.field)))
+      Consistency.messages ++= FastMessaging.message(errorNode, s"target fields are not allowed to have duplicates")
```

Comment:
This will be the error message for the user, right? Could you make that a bit more explicit as to what is and isn't allowed?

Reply:
Is the new version ok?

Reply:
Yes, thanks!